daffodil-japi/src/main/scala/org/apache/daffodil/japi/Daffodil.scala

```diff
@@ -1155,3 +1156,23 @@ class DaffodilUnparseContentHandler private[japi] (sContentHandler: SDaffodilUnp
   override def skippedEntity(name: String): Unit =
     contentHandler.skippedEntity(name)
 }
+
+/**
+ * Returns the EntityResolver used by Daffodil to resolve import/include
+ * schemaLocations.
+ *
+ * DFDLCatalogResolver uses an org.apache.xml.resolver.Catalog/CatalogManager to
+ * attempt to resolve namespaces and systemId's. If we are unable to resolve a
+ * file with the Catalog, we attempt to resolve the file using the systemId path
+ * from the schemaLocation attribute. If this fails we throw a
+ * SAXParseException.
+ *
+ * The DFDLCatalogResolver isn't thread safe, but it also is expensive and stateful,
+ * so we use ThreadLocal to only create one instance per thread.
+ */
+object DaffodilXMLEntityResolver {
+  def get = DFDLCatalogResolver.get
```

Comment:
I don't think we should return this. No one needs access to a `DFDLCatalogResolver` and it's capabilities, they just need one of the other kind of revolvers. If we do find someone needs access to this, we can consider making it public. But until then let's keep that implementation private. It's much easier to make something public that's private than to make something private that's public.